api/services/user_program_service.py

```diff
@@ -15,3 +13,12 @@ def set_init_data(self, jsonData):
         self.user_phone = helpers.sanitize_phone_string(user_phone)
         user = models.User.query.get_by_phone(self.user_phone)
         self.user_id = user.id
+
+    def mark_user_program_as_completed(self, JsonData):
+        self.set_init_data(JsonData)
+        self.user_program_data = models.UserProgram.get_by_user_id(self.user_id)
+        if self.user_program_data:
+            self.user_program_data.status = (
+                models.UserProgram.UserProgramStatus.COMPLETE
+            )
```

Comment:
While marking the user_program status as **completed** we need to update the end_date of the user_program also.